File: library/src/main/java/com/ycbjie/library/http/ApiException.java
package com.ycbjie.library.http;

/**
 * @author yc
 */
public class ApiException extends Exception {

    private int code;
    private String displayMessage;

    public ApiException(Throwable throwable, int code) {
        super(throwable);
        this.code = code;
    }

    public void setDisplayMessage(String displayMessage) {
        this.displayMessage = displayMessage;
    }

    public String getDisplayMessage() {
        return displayMessage;
    }

    public int getCode() {
        return code;
    }
}

File: android/src/main/java/com/ycbjie/android/network/ApiException.kt
package com.ycbjie.android.network


/**
 * 异常处理
 */
class ApiException : Exception {

    private var code: Int = 0

    constructor(message: String?, cause: Throwable?, code: Int) : super(message, cause) {
        this.code = code
    }

}